### PR TITLE
String Type Fixes

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
This is a PR to this PR: https://github.com/OpenMined/PySyft/pull/4532

## Affected Dependencies
None

## How has this been tested?
Locally in Python 3.6-3.8 and CI

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/syft/lib/python/string.py [hunk 1]
```diff
@@ -155,9 +156,11 @@ def islower(self) -> PyPrimitive:
 
     @syft_decorator(typechecking=True, prohibit_args=True)
     def isascii(self) -> PyPrimitive:
-        # Got mypy issues saying that isascii is not defined in superclass
-        res = self.data.isascii()
-        return PrimitiveFactory.generate_primitive(value=res)
+        if sys.version_info >= (3, 7):
+            res = self.data.isascii()
```
Comment (gmuraru) [inline src/syft/lib/python/string.py:160]: Q: Could you try here: ```super().isascii()```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -214,7 +214,7 @@ jobs:
           pip install torch==$TORCH_VERSION
           pip install -r requirements.txt
           pip install -e .
-          pytest -k "test_all_allowlisted" --cov-fail-under 0
+          pytest -m slow --cov-fail-under 0
```
Comment (Jasopaum) [inline .github/workflows/tests.yml:217]: nit but should we rename the job if we don't perform only allowlist tests anymore here?

File: proto/lib/python/none.proto [hunk 3]
```diff
@@ -5,5 +5,5 @@ import "proto/core/common/common_object.proto";
 
 
 message None {
-    syft.core.common.UID id_at_location = 2;
+    syft.core.common.UID id = 2;
```
Comment (Jasopaum) [inline proto/lib/python/none.proto:8]: nit: syft.core.common.UID id = 2 -> 1?

File: src/syft/lib/python/string.py (hunk 1 above)
Comment (madhavajay) [reply src/syft/lib/python/string.py:160]: Yup, fixing it now!

Comment (madhavajay) [reply src/syft/lib/python/string.py:160]: Weird...
```
src/syft/lib/python/string.py:160: error: "isascii" undefined in superclass
Found 1 error in 1 file (checked 194 source files)
```

There was a comment here about this before I think.

Comment (madhavajay) [reply src/syft/lib/python/string.py:160]: Im going to add # type: ignore since this seems silly.

Comment (gmuraru) [reply src/syft/lib/python/string.py:160]: Yes - I added the comment because I do not know how that ```isascii``` is working :\

**Review by Jasopaum — APPROVED** [review]

**Review by tudorcebere — APPROVED** [review]
LGTM!

**Review by gmuraru — APPROVED** [review]

**Pull request merged**